sdk/schemaregistry/Microsoft.Azure.Data.SchemaRegistry.ApacheAvro/tests/SchemaRegistryAvroObjectSerializerLiveTests.cs

```diff
@@ -48,6 +48,49 @@ public async Task CanSerializeAndDeserialize()
             Assert.AreEqual(42, readEmployee.Age);
         }
 
+        [RecordedTest]
+        public async Task CanSerializeAndDeserializeWithCompatibleSchema()
+        {
+            var client = CreateClient();
+            var groupName = TestEnvironment.SchemaRegistryGroup;
+            var employee = new Employee_V2 { Age = 42, Name = "Caketown", City = "Redmond" };
+
+            var encoder = new SchemaRegistryAvroEncoder(client, groupName, new SchemaRegistryAvroObjectEncoderOptions() { AutoRegisterSchemas = true });
+            (string schemaId, BinaryData data) = await encoder.EncodeAsync(employee, typeof(Employee_V2), CancellationToken.None);
+
+            // deserialize using the old schema, which is forward compatible with the new schema
+            // if you swap the old schema and the new schema in your mind, then this can also be thought as a backwards compatible test
+            var deserializedObject = await encoder.DecodeAsync(data, schemaId, typeof(Employee), CancellationToken.None);
+            var readEmployee = deserializedObject as Employee;
+            Assert.IsNotNull(readEmployee);
+            Assert.AreEqual("Caketown", readEmployee.Name);
+            Assert.AreEqual(42, readEmployee.Age);
```

Comment:
can we also somehow check that the city field does not exist?

Reply:
its strongly typed, so it doesn't 😄